File: backend/core-api/src/main/java/ai/govbiz/core/dto/support/SupportProgramSearchResponse.java
package ai.govbiz.core.dto.support;

import java.util.List;

import ai.govbiz.core.service.SupportProgramSearchResult;

public record SupportProgramSearchResponse(
        String query,
        List<SupportProgramResponse> programs
) {

    public static SupportProgramSearchResponse from(SupportProgramSearchResult result) {
        return new SupportProgramSearchResponse(
                result.query(),
                result.programs().stream().map(SupportProgramResponse::from).toList());
    }
}

File: backend/core-api/src/main/kotlin/ai/govbiz/core/dto/support/SupportProgramSearchResponse.kt
package ai.govbiz.core.dto.support

import ai.govbiz.core.service.SupportProgramSearchResult

data class SupportProgramSearchResponse(
    val query: String,
    val programs: List<SupportProgramResponse>,
) {
    companion object {
        @JvmStatic
        fun from(result: SupportProgramSearchResult): SupportProgramSearchResponse =
            SupportProgramSearchResponse(
                query = result.query,
                programs = java.util.List.copyOf(
                    result.programs.map(SupportProgramResponse::from),
                ),
            )
    }
}
